### PR TITLE
Streams

### PR DESCRIPTION
**Pull request by brozorec**
Add methods that return `Observable`s on which we can subsribe.

### CODE REVIEW

File: packages/sdk/src/entities/BorrowingVault.ts [hunk 1]
```diff
@@ -157,10 +180,12 @@ export class BorrowingVault {
   /**
    * Creates a connection by setting an rpc provider.
    *
-   * @param configParams - {@link ConfigParams} object with infura and alchemy ids
+   * @param configParams - {@link ChainConfigParams} object with infura and alchemy ids
    */
-  setConnection(configParams: ConfigParams): BorrowingVault {
-    this.rpcProvider = Config.rpcProviderFrom(configParams, this.chainId);
+  setConnection(configParams: ChainConfigParams): BorrowingVault {
```
Comment (doliG) [inline packages/sdk/src/entities/BorrowingVault.ts:185]: Maybe `ChainConfigParam` could be shortened with `ChainConfig` (almost the same meaning 🤔 )

File: packages/sdk/src/entities/BorrowingVault.ts [hunk 2]
```diff
@@ -275,6 +298,30 @@ export class BorrowingVault {
...
 
+  /**
+   * Returns a stream of deposit and borrow balances for an account.
+   *
+   * @param account - user address, wrapped in {@link Address}
+   * @throws if {@link setConnection} was not called beforehand
+   */
+  getBalancesStream(account: Address): Observable<AccountBalances> {
+    invariant(this.contract && this.wssProvider, 'Connection not set!');
+    const filters = [
```
Comment (doliG) [inline packages/sdk/src/entities/BorrowingVault.ts:309]: We don't need `flashclose | liquidate` in our filters ?

File: packages/sdk/src/entities/ChainConnection.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { ChainId } from '../enums';
+import { ChainConfigParams } from '../types';
+
+type ChainConnectionParams = {
+  rpcProvider: StaticJsonRpcProvider;
+  wssProvider?: WebSocketProvider;
+};
+
+export class ChainConnection {
+  private static _config: {
```
Comment (doliG) [inline packages/sdk/src/entities/ChainConnection.ts:21]: can be converted to Record 🤔

File: packages/sdk/test/entities/BorrowingVault.test.ts [hunk 4]
```diff
@@ -8,14 +8,17 @@ import { Address, BorrowingVault } from '../../src/entities';
...
+  ChainConfigParams,
   DepositParams,
   PermitParams,
   RouterActionParams,
 } from '../../src/types';
 
-const config: ConfigParams = {
+const config: ChainConfigParams = {
   infuraId: process.env.INFURA_ID ?? '',
+  alchemy: {
```
Comment (doliG) [inline packages/sdk/test/entities/BorrowingVault.test.ts:19]: alchemy should be also passed to sdk when constructing the classe ?

**Review by doliG — APPROVED** [review]

File: packages/sdk/src/entities/ChainConnection.ts (hunk 3 above)
Comment (brozorec) [reply packages/sdk/src/entities/ChainConnection.ts:21]: `[chainId in ChainId]?` allows optionality while it's not the case with Record afaik

File: packages/sdk/src/entities/BorrowingVault.ts (hunk 2 above)
Comment (brozorec) [reply packages/sdk/src/entities/BorrowingVault.ts:309]: good catch about liquidate

File: packages/sdk/test/entities/BorrowingVault.test.ts (hunk 4 above)
Comment (brozorec) [reply packages/sdk/test/entities/BorrowingVault.test.ts:19]: done

File: packages/sdk/src/entities/BorrowingVault.ts (hunk 1 above)
Comment (brozorec) [reply packages/sdk/src/entities/BorrowingVault.ts:185]: done

**Pull request merged**